source/features/preserve-text-messages.js

```diff
@@ -0,0 +1,99 @@
+import pick from 'lodash.pick';
+
+import {
+	getFromLocalStorage,
+	setToLocalStorage
+} from '../libs/utils';
+
+export async function removeMessages(refinedMsgIdsMaker) {
+	const {messages: savedMessages} = await getFromLocalStorage('messages');
+
+	if (!savedMessages) {
+		return;
+	}
+
+	const idsOfMsgToReSave = refinedMsgIdsMaker(savedMessages);
+
+	const updatedMessages = {
+		messages: pick(savedMessages, idsOfMsgToReSave)
+	};
+
+	setToLocalStorage(updatedMessages);
+}
+
+export function idsOfNonEmptyMsgs(savedMessages) {
+	return Object.keys(savedMessages)
+		.filter(id => !isEmptyMsgInput(savedMessages[id]));
+}
+
+export function idsOfNotDeletedMsgs(savedMessages) {
+	return Object.keys(savedMessages)
+		.filter(id => id !== getConversationId());
+}
+
+// See: https://gist.github.com/al3x-edge/1010364
+function setCursorToEnd(contentEditableElement) {
+	const range = document.createRange();
+	const selection = window.getSelection();
+	range.selectNodeContents(contentEditableElement);
+	range.collapse(false);
+	selection.removeAllRanges();
+	selection.addRange(range);
+}
+
+function getConversationId() {
+	return document.querySelector('.DMConversation').dataset.threadId;
+}
+
+function getMessageContainer() {
+	return $('#tweet-box-dm-conversation');
+}
+
+function isEmptyMsgInput(message) {
+	const messageEl = document.createElement('div');
+	messageEl.innerHTML = message;
```

Comment:
Bear me my naiveness, could you help me understand how is this unsafe, so that I could learn about it and find a way to fix it?

Reply:
https://stackoverflow.com/questions/30661497/xss-prevention-and-innerhtml

Reply:
@sindresorhus , since we are only using the `textContent ` property of that dom node for check for an empty message, shouldn't we be okay using this without sanitisation ?

Reference — https://stackoverflow.com/a/44018056/5106072

Reply:
Should I use a package like [dompurify](https://www.npmjs.com/package/dompurify) to fix this?

Reply:
@filipekiss ?